source/common/network/connection_impl.cc

```diff
@@ -697,16 +697,17 @@ ClientConnectionImpl::ClientConnectionImpl(
       file_event_->activate(Event::FileReadyType::Write);
       return;
     }
-    const Network::Address::Instance* source_to_use = source_address.get();
-    if (socket_->localAddress()) {
-      source_to_use = socket_->localAddress().get();
-    }
 
-    if (source_to_use != nullptr) {
-      const Api::SysCallIntResult result = source_to_use->bind(ioHandle().fd());
+    const Network::Address::InstanceConstSharedPtr* source = &source_address;
+
+    if (socket_->localAddress())
```

Comment:
braces around all if statements (tidy will check this)

Reply:
Fixed!